vulnerabilities/pipelines/huawei_bulletin_importer.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+import logging
+from datetime import datetime
+from datetime import timezone
+from typing import Iterable
+from typing import Tuple
+
+import requests
+from bs4 import BeautifulSoup
+from packageurl import PackageURL
+from univers.version_range import VersionRange
+from univers.versions import GenericVersion
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipeline
+from vulnerabilities.severity_systems import GENERIC
+
+logger = logging.getLogger(__name__)
+
+
+def extract_version(version_str: str) -> Tuple[str, str]:
+    """
+    Gets the version type and number from a messy Huawei version string.
+    Returns tuple of (type, version) - returns empty type if cant parse
+    """
+    version_str = version_str.lower().strip()
+    if version_str.startswith("harmonyos"):
+        return "harmonyos", version_str.split("harmonyos")[-1].strip()
+    elif version_str.startswith("emui"):
+        return "emui", version_str.split("emui")[-1].strip()
+    return "", version_str
+
+
+class HuaweiBulletinImporterPipeline(VulnerableCodeBaseImporterPipeline):
+    """
+    Scrapes security bullitin data from Huawei's website:
+    https://consumer.huawei.com/en/support/bulletin/
+    """
+
+    pipeline_id = "huawei_bulletin_importer"
+    spdx_license_expression = "LicenseRef-Terms-Of-Use"
+    license_url = "https://consumer.huawei.com/en/legal/terms-of-use/"
```

Comment:
This is not the license for the security bulletin. We should leave this empty for now and reach out to them later to clarify the license for the security advisory.